src/olympia/addons/indexers.py

```diff
@@ -119,18 +143,33 @@ def extract_document(cls, obj):
                 # Fake wide compatibility for search tools and personas.
                 min_, max_ = 0, version_int('9999')
             data['appversion'][app.id] = dict(min=min_, max=max_)
-        try:
-            data['has_version'] = obj._current_version is not None
-        except ObjectDoesNotExist:
-            data['has_version'] = None
-
         data['authors'] = [a.name for a in obj.listed_authors]
+        # Double the boost if the add-on is public.
+        if obj.status == amo.STATUS_PUBLIC and 'boost' in data:
+            data['boost'] = max(data['boost'], 1) * 4
         # We go through attach_categories and attach_tags transformer before
         # calling this function, it sets category_ids and tag_list.
         data['category'] = getattr(obj, 'category_ids', [])
         if obj.current_version:
+            data['current_version'] = {
+                'id': obj.current_version.pk,
+                'files': [{
+                    'id': file_.id,
+                    'created': file_.created,
+                    'filename': file_.filename,
+                    'hash': file_.hash,
+                    'platform': file_.platform,
+                    'size': file_.size,
+                    'status': file_.status,
+                } for file_ in obj.current_version.all_files],
+                'reviewed': obj.current_version.reviewed,
+                'version': obj.current_version.version,
+            }
+            data['has_version'] = True
             data['platforms'] = [p.id for p in
                                  obj.current_version.supported_platforms]
+        else:
+            data['has_version'] = None
```

Comment:
I think it's because `null` in ES is optimized, the value is not included in the index. Omitting it entirely would work the same way.